test/functional/legacy/057_spec.lua

```diff
@@ -0,0 +1,973 @@
+      abc
+      ab
+      a
+      a321
+      a123
+      a122
+      b321
+      b123
+      c123d
+       123b
+      c321d
+      b322b
+      b321
+      b321b
+      
+      
+      t16: numeric, sort on first character
+      abc
+      ab
+      a
+      a321
+      a123
+      a122
+      b321
+      b123
+      c123d
+       123b
+      c321d
+      b322b
+      b321
+      b321b
+      
+      
+      t17: alpha, skip past first digit
+      abc
+      ab
+      a
+      a321
+      a123
+      a122
+      b321
+      b123
+      c123d
+       123b
+      c321d
+      b322b
+      b321
+      b321b
+      
+      
+      t18: alpha, sort on first digit
+      abc
+      ab
+      a
+      a321
+      a123
+      a122
+      b321
+      b123
+      c123d
+       123b
+      c321d
+      b322b
+      b321
+      b321b
+      
+      
+      t19: numeric, skip past first digit
+      abc
+      ab
+      a
+      a321
+      a123
+      a122
+      b321
+      b123
+      c123d
+       123b
+      c321d
+      b322b
+      b321
+      b321b
+      
+      
+      t20: numeric, sort on first digit
+      abc
+      ab
+      a
+      a321
+      a123
+      a122
+      b321
+      b123
+      c123d
+       123b
+      c321d
+      b322b
+      b321
+      b321b
+      
+      
+      t21: alpha, skip past first 2 digits
+      abc
+      ab
+      a
+      a321
+      a123
+      a122
+      b321
+      b123
+      c123d
+       123b
+      c321d
+      b322b
+      b321
+      b321b
+      
+      
+      t22: numeric, skip past first 2 digits
+      abc
+      ab
+      a
+      a321
+      a123
+      a122
+      b321
+      b123
+      c123d
+       123b
+      c321d
+      b322b
+      b321
+      b321b
+      
+      
+      t23: hexadecimal, skip past first 2 digits
+      abc
+      ab
+      a
+      a321
+      a123
+      a122
+      b321
+      b123
+      c123d
+       123b
+      c321d
+      b322b
+      b321
+      b321b
+      
+      
+      t24: alpha, sort on first 2 digits
+      abc
+      ab
+      a
+      a321
+      a123
+      a122
+      b321
+      b123
+      c123d
+       123b
+      c321d
+      b322b
+      b321
+      b321b
+      
+      
+      t25: numeric, sort on first 2 digits
+      abc
+      ab
+      a
+      a321
+      a123
+      a122
+      b321
+      b123
+      c123d
+       123b
+      c321d
+      b322b
+      b321
+      b321b
+      
+      
+      t26: hexadecimal, sort on first 2 digits
+      abc
+      ab
+      a
+      a321
+      a123
+      a122
+      b321
+      b123
+      c123d
+       123b
+      c321d
+      b322b
+      b321
+      b321b
+      
+      
+      t27: wrong arguments
+      abc
+      ab
+      a
+      a321
+      a123
+      a122
+      b321
+      b123
+      c123d
+       123b
+      c321d
+      b322b
+      b321
+      b321b
+      
+      
+      t28: done
+      ]=])
+
+    --execute('so small.vim')
+
+    source([[
```

Comment:
I think it would be better to use `execute()` here (AFAIR you can execute multiple commands after one another if you chain them with commas, like `execute('/.../', '/...')`. `source` will write a temporary file, which is a bit of overhead I think we can avoid here.